lib/beaker-pe/install/pe_utils.rb

```diff
@@ -395,7 +397,33 @@ def fetch_pe_on_unix(host, opts)
                 command_file_push = "curl -L #{curlopts}#{path}/"
               end
               on host, "cd #{host['working_dir']}; #{command_file_push}#{filename}#{extension} | #{unpack}"
+              gpg_key_overwrite(host, 'tarball')
+            end
+          end
+        end
 
+        #PE-32680, GPG expired on older PE versions, need to update with a new GPG key on the primary server
+        #only affects PE versions 2019.8.4 and earlier, and only needed for debian, ubuntu, and sles agent platforms
+        # @param [Host] host to see if we need to update the gpg key
+        # @param [String] location of the GPG key we intend to overwrite
+        # If someone is using this gem and is not on the PE private network, they need to download the new private GPG key and host
+        # it somewhere, then set the URL as an enviromental variable GPG_URL.
+        def gpg_key_overwrite(host, location)
+          GPG_URL = ENV['GPG_URL'] || 'https://artifactory.delivery.puppetlabs.net/artifactory/generic__local/extended_gpg_key.asc'
+          case location
+          when 'tarball'
+            path_to_gpg_key = "#{host['working_dir']}/#{host['dist']}/packages/GPG-KEY-puppet"
+          when 'pe_repo'
+            path_to_gpg_key = '/opt/puppetlabs/puppet/modules/pe_repo/files/GPG-KEY-puppet'
```

Comment:
We might need this to go into `/opt/puppetlabs/server/data/environments/enterprise/modules/pe_repo` as well to account for the tests where we do `puppet infra configure`, which I think would end up putting the old key back maybe?

Reply:
👍

Reply:
Alright pushed up the change and it is now updating both places: 
```    
rational-spy.delivery.puppetlabs.net (centos7-64-1) 20:56:07$ rm -f /opt/puppetlabs/puppet/modules/pe_repo/files/GPG-KEY-puppet

    rational-spy.delivery.puppetlabs.net (centos7-64-1) executed in 0.08 seconds

    rational-spy.delivery.puppetlabs.net (centos7-64-1) 20:56:07$ curl https://artifactory.delivery.puppetlabs.net/artifactory/generic__local/extended_gpg_key.asc --output /opt/puppetlabs/puppet/modules/pe_repo/files/GPG-KEY-puppet
        % Total    %       Received % Xferd  A      verage Speed   Time    Time     Tim      e  Current
                                             Dload  Uplo      ad   Total   Spent    Left        Speed
        0     0          0     0    0     0            0      0 --:--:-- -      -:--:-- --:--:      --     0      100  7758  100  7758    0     0  66879            0 --:--:-- --:--:-- --:--:-- 66879

    rational-spy.delivery.puppetlabs.net (centos7-64-1) executed in 0.20 seconds

    rational-spy.delivery.puppetlabs.net (centos7-64-1) 20:56:07$ rm -f /opt/puppetlabs/server/data/environments/enterprise/modules/pe_repo/files/GPG-KEY-puppet

    rational-spy.delivery.puppetlabs.net (centos7-64-1) executed in 0.08 seconds

    rational-spy.delivery.puppetlabs.net (centos7-64-1) 20:56:07$ curl https://artifactory.delivery.puppetlabs.net/artifactory/generic__local/extended_gpg_key.asc --output /opt/puppetlabs/server/data/environments/enterprise/modules/pe_repo/files/GPG-KEY-puppet
```